File: 1768.merge-strings-alternately/solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder strBuilder = new StringBuilder();
        int i = 0;
        int j = 0;
        int N = word1.length();
        int M = word2.length();
        while ((i < N) || (j < M)) {
            if (i < N) {
                strBuilder.append(word1.charAt(i));
                i++;
            }
            if (j < M) {
                strBuilder.append(word2.charAt(j));
                j++;
            }
        }
        return strBuilder.toString();
    }
}

File: 1768.merge-strings-alternately/solution.ts
function mergeAlternately(word1: string, word2: string): string {
  let buffer: string[] = [];
  let i = 0;
  let j = 0;
  while (i < word1.length || j < word2.length) {
      if (i < word1.length) {
          buffer.push(word1[i]);
          i++;
      }
      if (j < word2.length) {
          buffer.push(word2[j]);
          j++;
      }
  }
  return buffer.join('');
};